### PR TITLE
Consolidate visit show page

### PR DESCRIPTION
**Pull request by alan**
This makes the visit view page be similar to the completed step when the request
hasn't been processed by including general information about the visit (while
keeping the cancel button).

Updates the view for when the visit has been booked to include all of
the general information as well as approved / rejected visitors.

Finally, it fixes the issue of resubmitting the form if the completed step was
refreshed.

### CODE REVIEW

**Comment by alan** [comment]
Depends on https://github.com/ministryofjustice/prison-visits-2/pull/355 and https://github.com/ministryofjustice/prison-visits-2/pull/356

File: app/controllers/booking_requests_controller.rb [hunk 1]
```diff
@@ -2,30 +2,48 @@ class BookingRequestsController < ApplicationController
...
-    Instrumentation.append_to_log booking_step_rendered: processor.step_name
+
+    instrument_booking_request(step_name: @step_name)
 
     respond_to do |format|
       format.html { render processor.step_name }
     end
   end
 
   def create
```
Comment (alan) [inline app/controllers/booking_requests_controller.rb]: The refactor here was necessary as it broke rubocop rules

File: config/locales/en/views.yml [hunk 2]
```diff
@@ -303,18 +303,27 @@ en:
       yes_cancel: Yes, I want to cancel this visit
```
Comment (alan) [inline config/locales/en/views.yml]: Any ideas on how to manage the translations? We have most of this text localised from the old completed step (which I haven't removed in this PR)

File: app/controllers/booking_requests_controller.rb (hunk 1 above)
Comment (mloughran) [inline app/controllers/booking_requests_controller.rb]: We agreed to disagree on this refactor (both pulling out assigning processor and the instrumentation). In my view this is unnecessary abstraction, which makes the code more complex.

**Comment by alan** [comment]
Depends on ministryofjustice/prison-visits-2#359

**Comment by mloughran** [comment]
This is now ready to merge from my POV, pending the related PR on PVB2. I have dealt with the translations (see 31c8fdb).

**Pull request merged**